sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
I'm sorry, I meant in symengine.py. Cython works a little different than regular Python classes. I'l check whether `__new__` thing works. If so, then `Add(x + x)` can work.